### DISCUSSION TITLE
Diligent's opinions on WebGPU's WGSL shading lauguange support?

### DISCUSSION THREAD

**Discussion by WangHoi** [DiligentGraphics/DiligentEngine#302] [General]
- Will it get native supported
- Or indirectly supported by Google's translator to spirv
- Or Diligent provide a HLSL2WGSL onverter ？

**Comment by MikhailGorobets** [comment]
WGSL should work with the WebGPU backend. WGSL is a terrible shader language; if I were you, I would use HLSL instead. 
WebGPU is primarily intended for the web platform, but everything should work natively(by Dawn) as well.

**Comment by WangHoi** [comment]
Thank you, sticky to HLSL now.

But i really wish Google can pull more people to make a Chrome-DevTools quality shader debugger.

**Comment by TheMostDiligent** [comment]
WGSL can be directly passed to Diligent.
Use [SHADER_SOURCE_LANGUAGE_WGSL](https://github.com/DiligentGraphics/DiligentCore/blob/88c3cb6ab5209a8b47f6407f63d9a83ff7cbcd56/Graphics/GraphicsEngine/interface/Shader.h#L82)
